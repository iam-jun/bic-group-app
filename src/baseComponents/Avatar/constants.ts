import { dimension, spacing } from '~/theme';

export const AVATAR_STYLES = {
  tiny: {
    width: dimension.avatarSizes.tiny,
    height: dimension.avatarSizes.tiny,
    borderRadius: spacing.borderRadius.small,
  },
  xSmall: {
    width: dimension.avatarSizes.xSmall,
    height: dimension.avatarSizes.xSmall,
    borderRadius: spacing.borderRadius.small,
  },
  small: {
    width: dimension.avatarSizes.small,
    height: dimension.avatarSizes.small,
    borderRadius: spacing.borderRadius.base,
  },
  base: {
    width: dimension.avatarSizes.base,
    height: dimension.avatarSizes.base,
    borderRadius: spacing.borderRadius.base,
  },
  medium: {
    width: dimension.avatarSizes.medium,
    height: dimension.avatarSizes.medium,
    borderRadius: spacing.borderRadius.large,
  },
  large: {
    width: dimension.avatarSizes.large,
    height: dimension.avatarSizes.large,
    borderRadius: spacing.borderRadius.large,
  },
  xLarge: {
    width: dimension.avatarSizes.xLarge,
    height: dimension.avatarSizes.xLarge,
    borderRadius: spacing.borderRadius.large,
  },
};

export const PRIVACY_ICON_VIEW_SIZES = {
  tiny: 10,
  xSmall: 12,
  small: 12,
  base: 20,
  medium: 24,
  large: 28,
  xLarge: 32,
};

export const PRIVACY_ICON_SIZES = {
  tiny: 6,
  xSmall: 8,
  small: 8,
  base: 12,
  medium: 16,
  large: 20,
  xLarge: 24,
};

export const ACTION_CONTAINER_SIZES = 20;

export const ACTION_ICON_SIZES = 12;
