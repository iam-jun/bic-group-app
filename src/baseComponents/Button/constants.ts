import { TextVariant } from '~/beinComponents/Text';
import { IObject } from '~/interfaces/common';

export const BUTTON_SIZES = {
  large: 44,
  medium: 36,
  small: 28,
};

export const TEXT_SIZES: IObject<TextVariant> = {
  large: 'buttonL',
  medium: 'buttonM',
  small: 'buttonS',
};

export const ICON_SIZES = {
  large: 18,
  medium: 18,
  small: 14,
};

export const BUTTON_PADDING = {
  large: 16,
  medium: 12,
  small: 8,
};
