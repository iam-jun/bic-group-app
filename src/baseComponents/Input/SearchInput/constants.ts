export const SEARCH_INPUT_SIZES = {
  large: 36,
  medium: 32,
  small: 28,
};
