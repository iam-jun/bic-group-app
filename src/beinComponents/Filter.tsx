import { ExtendedTheme, useTheme } from '@react-navigation/native';
import React, { useCallback, useMemo, useState } from 'react';
import {
  View,
  StyleSheet,
  ScrollView,
  StyleProp,
  ViewStyle,
  TouchableOpacity,
  Dimensions,
} from 'react-native';
import { PanGestureHandler } from 'react-native-gesture-handler';
import Animated, {
  Easing,
  interpolate,
  useAnimatedGestureHandler,
  useAnimatedStyle,
  useDerivedValue,
  useSharedValue,
  withTiming,
} from 'react-native-reanimated';

import spacing from '~/theme/spacing';
import Icon from '../baseComponents/Icon';
import Text from '../baseComponents/Text';

const { width: screenWidth } = Dimensions.get('window');
export interface FilterProps {
  translateX: Animated.SharedValue<number>;
  data?: {id: number; text: string; icon?: string; type: string}[];
  activeIndex?: number;
  filterRef?: React.Ref<ScrollView>;
  testID?: string;
  itemTestID?: string;
  style?: StyleProp<ViewStyle>;
  showItemBorder?: boolean;
  activeColor?: string;
  activeBackgroundColor?: string;
  inactiveColor?: string;

  onPress: (item: any, index: number) => void;
}

const FilterComponent: React.FC<FilterProps> = ({
  testID,
  style,
  data = [],
  itemTestID,
  onPress,
  activeIndex = 0,
  translateX,
  showItemBorder = true,
  activeColor,
  activeBackgroundColor,
  inactiveColor,
}: FilterProps) => {
  const theme: ExtendedTheme = useTheme();
  const styles = useMemo(
    () => createStyle(theme), [theme, style],
  );

  const [measurements, setMeasurements] = useState<any[]>(new Array(data?.length).fill({}));

  const _onPress = useCallback(
    (
      item: any, index: number,
    ) => {
      onPress?.(
        item, index,
      );
    },
    [onPress],
  );

  const panGestureValue = useSharedValue(0);
  const offsetValue = useSharedValue(0);
  const clampedTranslateX = useDerivedValue(() => {
    const lastMeasurements = measurements[measurements.length - 1];
    const MAX_TRANSLATE_X = -(
      (lastMeasurements?.x || 0)
        + (lastMeasurements?.width || 0)
        + spacing.margin.small
        - screenWidth
    ) || 0;
    if (MAX_TRANSLATE_X > 0) return 0;
    return Math.max(
      Math.min(
        panGestureValue.value, 0,
      ), MAX_TRANSLATE_X,
    );
  });

  const renderItem = (
    item: any, index: number,
  ) => {
    const isSelected = index === activeIndex;
    const itemStyle = [styles.itemContainer, showItemBorder ? styles.itemBorder : {}];
    const defaultTextColor = theme.colors.neutral80;
    const textColor = isSelected ? activeColor || defaultTextColor : inactiveColor || defaultTextColor;
    return (
      <View
        style={styles.itemView}
        key={`${itemTestID || 'item_filter'}_${item?.text}`}
        onLayout={(event) => {
          const { x, width, height } = event?.nativeEvent?.layout || {};
          if (
            measurements[index]?.x !== x
            || measurements[index]?.width !== width
          ) {
            measurements[index] = { x, width, height };
            setMeasurements([...measurements]);
          }
        }}
      >
        <TouchableOpacity
          activeOpacity={0.25}
          style={itemStyle}
          testID={`${itemTestID || 'item_filter'}_${item.id}`}
          onPress={() => {
            _onPress(
              item, index,
            );
          }}
        >
          {!!item?.icon && (
            <Icon
              icon={item.icon}
              size={24}
              tintColor={
                isSelected ? theme.colors.purple60 : theme.colors.gray50
              }
              style={styles.icon}
            />
          )}
          <Text color={textColor} variant={isSelected ? 'bodyMMedium' : 'bodyM'} useI18n>
            {item.text}
          </Text>
        </TouchableOpacity>
      </View>
    );
  };

  const panGestureEvent = useAnimatedGestureHandler({
    onActive(event: any) {
      if (panGestureValue.value > 0) {
        offsetValue.value = 0;
        panGestureValue.value = event.translationX;
      } else {
        panGestureValue.value = event.translationX + offsetValue.value;
      }
    },
    onFinish() {
      offsetValue.value = panGestureValue.value;
    },
  });

  const scrollViewContainerStyle = useAnimatedStyle(() => ({
    transform: [
      {
        translateX: withTiming(
          clampedTranslateX.value, {
            duration: 10,
            easing: Easing.bezier(
              0.25, 0.1, 0.25, 1,
            ),
          },
        ),
      },
    ],
  }));

  const activeStyle = useAnimatedStyle(() => ({
    transform: [
      {
        translateX: interpolate(
          translateX?.value,
          data?.map?.((
            _, index: number,
          ) => index * screenWidth),
          measurements.map((item: any) => {
            if (!item?.x) return 8;
            return item.x;
          }),
        ),
      },
    ],
    width: interpolate(
      translateX?.value,
      data?.map?.((
        _, index: number,
      ) => index * screenWidth),
      measurements.map((item: any) => item?.width || 60),
    ),
    height: measurements[0]?.height || 0,
    borderWidth: measurements[0]?.height ? 1 : 0,
  }));

  const itemActiveStyle = activeBackgroundColor ? {
    backgroundColor: activeBackgroundColor,
    borderColor: activeBackgroundColor,
  } : styles.itemActiveColor;

  return (
    <View testID={testID || 'filter'} style={[styles.container, style]}>
      <PanGestureHandler onGestureEvent={panGestureEvent}>
        <Animated.View style={[scrollViewContainerStyle]}>
          <View
            style={styles.row}
          >
            {data?.map?.(renderItem)}
            <Animated.View
              style={[
                { ...StyleSheet.absoluteFillObject },
                styles.itemActive,
                itemActiveStyle,
                activeStyle,
              ]}
            />
          </View>
        </Animated.View>
      </PanGestureHandler>
    </View>
  );
};

const createStyle = (theme: ExtendedTheme) => {
  const { colors } = theme;
  return StyleSheet.create({
    container: {
      paddingVertical: spacing.padding.base,
      backgroundColor: colors.white,
      borderTopWidth: 1,
      borderBottomWidth: 1,
      borderColor: colors.neutral5,
    },
    itemView: {
      padding: 0,
      margin: 0,
      marginLeft: spacing.margin.small,
    },
    itemContainer: {
      backgroundColor: colors.transparent,
      flexDirection: 'row',
      paddingVertical: spacing.padding.small,
      paddingHorizontal: spacing.padding.base,
    },
    itemBorder: {
      borderWidth: 1,
      borderRadius: 100,
      borderColor: colors.neutral5,
    },
    itemActive: {
      borderRadius: 100,
      zIndex: -1,
    },
    itemActiveColor: {
      borderColor: colors.gray40,
      backgroundColor: colors.gray40,
    },
    iconLeftStyle: { marginRight: spacing.margin.base },
    icon: {
      marginRight: spacing.margin.small,
    },
    row: {
      flexDirection: 'row',
      paddingLeft: spacing.padding.small,
    },
  });
};

const _Filter = React.forwardRef((
  props: FilterProps, ref?: React.Ref<ScrollView>,
) => (
  <FilterComponent filterRef={ref} {...props} />
));

const Filter = React.memo(_Filter);
Filter.whyDidYouRender = true;
export default Filter;
