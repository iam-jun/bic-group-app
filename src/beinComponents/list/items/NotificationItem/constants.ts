export const VERB = {
  POST: 'post',
  MENTION: 'mention',
};

export const COMMENT_TARGET = {
  POST: 'post',
  COMMENT: 'comment',
};

export const SAMPLE_ACTIVITY_1 = {
  actor: {
    created_at: '2022-01-14T04:58:45.776555Z',
    updated_at: '2022-03-01T08:18:56.458626Z',
    id: '20',
    data: {
      avatar:
        'https://bein-entity-attribute-stg.s3.ap-southeast-1.amazonaws.com/user/avatar/Avatar_Profile.png',
      fullname: 'Trần Nam Anh',
    },
  },
  foreign_id: 'reaction:19f416e1-ef1c-46b3-9f53-215345fbca88',
  id: 'e361c20a-9ac9-11ec-8080-80014d2f0ccf',
  notification_type: 7,
  origin: null,
  target: '',
  time: '2022-03-03T08:14:03.558145',
  verb: 'comment',
  mentions: {
    users: {},
    groups: {},
  },
};

export const SAMPLE_ACTIVITY_2 = {
  actor: {
    created_at: '2022-01-14T04:58:45.776555Z',
    updated_at: '2022-03-01T08:18:56.458626Z',
    id: '21',
    data: {
      avatar:
        'https://bein-entity-attribute-stg.s3.ap-southeast-1.amazonaws.com/user/avatar/Avatar_Profile.png',
      fullname: 'Huỳnh Phương Khanh',
    },
  },
  foreign_id: 'reaction:19f416e1-ef1c-46b3-9f53-215345fbca88',
  id: 'e361c20a-9ac9-11ec-8080-80014d2f0ccf',
  notification_type: 7,
  origin: null,
  target: '',
  time: '2022-03-03T08:14:03.558145',
  verb: 'comment',
  mentions: {
    users: {},
    groups: {},
  },
};
