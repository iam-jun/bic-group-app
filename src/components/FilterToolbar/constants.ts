import { PostType } from '~/interfaces/IPost';

export enum TypeFilter {
  All = 'ALL',
  Today = 'TODAY',
  Yesterday = 'YESTERDAY',
  LastSevenDays = 'LAST_SEVEN_DAYS',
  FromTo = 'FROM_TO'
}

export const itemFilter = [
  {
    key: TypeFilter.Today,
    text: 'home:newsfeed_search:today',
  },
  {
    key: TypeFilter.Yesterday,
    text: 'home:newsfeed_search:yesterday',
  },
  {
    key: TypeFilter.LastSevenDays,
    text: 'home:newsfeed_search:last_seven_days',
  },
  {
    key: TypeFilter.FromTo,
    text: 'home:newsfeed_search:specific_day',
  },
];

export const postTypeFilter = [
  {
    key: PostType.POST,
    text: 'home:newsfeed_search:filter_post',
  },
  {
    key: PostType.ARTICLE,
    text: 'home:newsfeed_search:filter_article',
  },
  {
    key: PostType.SERIES,
    text: 'home:newsfeed_search:filter_series',
  },
];
