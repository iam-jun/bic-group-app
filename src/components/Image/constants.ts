// image's width that BE supports
export const BeinImageSizes = [
  16,
  32,
  48,
  64,
  96,
  128,
  256,
  384,
  640,
  750,
  828,
  1080,
  1200,
  1920,
  2048,
  3840,
];
