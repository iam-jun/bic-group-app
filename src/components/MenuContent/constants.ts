export const defaultMenu = {
  canCopyLink: {
    // id: 4,
    // testID: 'post_view_menu.copy',
    // leftIcon: 'LinkHorizontal',
    // title: i18next.t('post:post_menu_copy'),
    // requireIsActor: false,
    // onPress: onPressCopyLink,
  },
};

export const MENU_KEYS = {
  EDIT: 'canEdit',
  EDIT_SETTING: 'canEditSetting',
  SAVE: 'canSave',
  COPY_LINK: 'canCopyLink',
  VIEW_REACTIONS: 'canViewReactions',
  VIEW_SERIES: 'canViewSeries',
  PIN_CONTENT: 'canPinContent',
  CREATE_QUIZ: 'canCreateQuiz',
  DELETE_QUIZ: 'canDeleteQuiz',
  EDIT_QUIZ: 'canEditQuiz',
  DELETE: 'canDelete',
  REPORT_CONTENT: 'canReportContent',
  REPORT_MEMBER: 'canReportMember',
  ENABLE_SPECIFIC_NOTIFICATIONS: 'enableSpecificNotifications',
};
