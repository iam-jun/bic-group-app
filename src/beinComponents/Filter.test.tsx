import * as React from 'react';
import { cleanup, fireEvent } from '@testing-library/react-native';

import Filter from './Filter';
import { renderWithRedux } from '~/test/testUtils';
import { communityMenuData } from '~/constants/communityMenuData';

afterEach(cleanup);

describe('Filter component', () => {
  it('renders correctly', () => {
    const onPress = jest.fn();
    const rendered = renderWithRedux(
      <Filter data={communityMenuData} selectedIndex={0} onPress={onPress} />,
    ).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('should called onPress prop', () => {
    const onPress = jest.fn();
    const selectedIndex = 0;
    const rendered = renderWithRedux(
      <Filter
        selectedIndex={selectedIndex}
        data={communityMenuData}
        onPress={onPress}
        testID="community_menu"
        itemTestID="item_community_data"
      />,
    );
    const itemComponent = rendered.getByTestId('item_community_data_2');
    expect(itemComponent).toBeDefined();
    fireEvent.press(itemComponent);
    expect(onPress).toHaveBeenCalled();
  });
});
