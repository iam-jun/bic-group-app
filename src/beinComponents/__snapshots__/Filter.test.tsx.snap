// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "borderBottomWidth": 1,
        "borderColor": "#E3E6F2",
        "borderTopWidth": 1,
        "paddingVertical": 12,
      },
      undefined,
    ]
  }
  testID="filter"
>
  <View
    collapsable={false}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    style={
      Array [
        Object {
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "margin": 0,
            "marginLeft": 8,
            "padding": 0,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(255,255,255,0)",
              "borderColor": "#E3E6F2",
              "borderRadius": 100,
              "borderWidth": 1,
              "flexDirection": "row",
              "opacity": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
          testID="item_filter_1"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#212745",
                  "fontFamily": "BeVietnamPro-Medium",
                  "fontSize": 15,
                },
                Object {},
                undefined,
              ]
            }
          >
            You've joined
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "margin": 0,
            "marginLeft": 8,
            "padding": 0,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(255,255,255,0)",
              "borderColor": "#E3E6F2",
              "borderRadius": 100,
              "borderWidth": 1,
              "flexDirection": "row",
              "opacity": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
          testID="item_filter_2"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#212745",
                  "fontFamily": "BeVietnamPro-Light",
                  "fontSize": 15,
                },
                Object {},
                undefined,
              ]
            }
          >
            You manage
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "margin": 0,
            "marginLeft": 8,
            "padding": 0,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "rgba(255,255,255,0)",
              "borderColor": "#E3E6F2",
              "borderRadius": 100,
              "borderWidth": 1,
              "flexDirection": "row",
              "opacity": 1,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
            }
          }
          testID="item_filter_3"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#212745",
                  "fontFamily": "BeVietnamPro-Light",
                  "fontSize": 15,
                },
                Object {},
                undefined,
              ]
            }
          >
            Discover
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "backgroundColor": "#A4A9B7",
              "borderColor": "#A4A9B7",
              "borderRadius": 100,
              "zIndex": -1,
            },
            Object {
              "borderWidth": 0,
              "height": 0,
              "transform": Array [
                Object {
                  "translateX": undefined,
                },
              ],
              "width": undefined,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
